src/main/java/dao/UserDaoImpl.java

```diff
@@ -44,4 +44,15 @@ public void add(User user) {
             throw new RuntimeException("Can't list all users", e);
         }
     }
+
+    @Override
+    public User get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("SELECT u FROM User u WHERE u.id = :id", User.class)
+                    .setParameter("id", id)
+                    .getSingleResult();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find user", e);
+        }
```

Comment:
checklist
"Don’t complicate get by id methods in your DAO, use session.get().
"